lib/assert.js

```diff
@@ -278,7 +278,13 @@ function expectedException(actual, expected) {
     // Ignore.  The instanceof check doesn't work for arrow functions.
   }
 
-  return expected.call({}, actual) === true;
+  try {
+    return expected.call({}, actual) === true;
+  } catch (e) {
+    // Ignore. If expected is a class constructor, an error will be thrown
+  }
```

Comment:
Does this mean that `assert.throws()` will now ignore all errors thrown by the validation function? (And are there semver implications to such a change? Wouldn't we want things like `ReferenceError` to bubble up to the user?)